skimage/util/_flood_fill_cy.pyx

```diff
@@ -0,0 +1,143 @@
+#cython: cdivision=True
+#cython: boundscheck=False
+#cython: nonecheck=False
+#cython: wraparound=False
+
+"""Cython code used in _flood_fill.py."""
+
+cimport numpy as cnp
+
+
+# Must be defined to use QueueWithHistory
+ctypedef Py_ssize_t QueueItem
+
+include "../morphology/_queue_with_history.pxi"
+
+
+ctypedef fused dtype_t:
+    cnp.uint8_t
+    cnp.uint16_t
+    cnp.uint32_t
+    cnp.uint64_t
+    cnp.int8_t
+    cnp.int16_t
+    cnp.int32_t
+    cnp.int64_t
+    cnp.float32_t
+    cnp.float64_t
+
+
+# Definition of flag values used for `flags` in _flood_fill & _fill_plateau
+cdef:
+    # Border value - do not cross!
+    unsigned char BORDER = 2
+    # Part of the flood fill
+    unsigned char FILL = 1
+    # Not checked yet
+    unsigned char UNKNOWN = 0
+
+
+cpdef inline void _flood_fill_equal(dtype_t[::1] image, 
+                                    unsigned char[::1] flags,
+                                    Py_ssize_t[::1] neighbor_offsets, 
+                                    Py_ssize_t start_index, 
+                                    dtype_t seed_value) nogil:
```

Comment:
Right now the `nogil` statement has no effect on this function when calling from Python. I'd recommend enclosing everything after line 64 with a `with nogil:` block. 
And I'm not sure how this would interact with the `cpdef`, `inline` and `nogil` in the functions signature so I'd change that one to `def` for now.